src/lbp.jl

```diff
@@ -0,0 +1,144 @@
+type UNIFORM_PARAMS
+	table::Dict{BitArray{1}, Int}
+	count::Int
+	non_uniform_pattern::BitArray{1}
+end
```

Comment:
Here's how you can tell:
```jl
julia> isleaftype(BitArray)
false

julia> isleaftype(BitArray{1})
true
```
As long as it's a leaf type, it doesn't have to be a parameter. Your `Dict` should be fine too.

Any special reason for the all-caps? How about `LBPatternDict` or `PatternCache` or something like that?